### DISCUSSION TITLE
Split does not always create smaller files

### DISCUSSION THREAD

**Discussion by petrbaranov** [pdfcpu/pdfcpu#704] [Q&A]
I noticed that `pdfcpu split path/to/file.pdf out 2` works differently depending on the input file. It always produces files containing correct number of pages, but their size is not always as expected. It happens that produces files have same size as the input file, though they contains only few pages (as expected) from the initial file.

What may cause such behavior and how to split PDF so that file size always corresponds to the content in the file? 

Thank you in advance,
Petr

**Comment by hhrutter** [comment]
The size issue may be related to the fact that some page attributes may be inherited.
It will be easier to address this if you can share maybe one or two small sample files.

**Comment by nicolasfranck** [comment]
I also had that problem in the past, but only with older pdf's. It turns out this was due to the way images were embedded in the past: they were added to pdf headers, and then referenced from the pdf pages, as a way to reuse the images. The problem with this behaviour is that when you split a pdf, the pdf splitter will copy the pdf headers into each new pdf. So every new pdf has all of the images of the original in its headers (so unvisible), but only uses one. e.g. if the original had 10 pages, then every new pdf has 10 images, and uses only one.

You'll have to rewrite the pdf into a new one, which will remove this old behaviour, and put every image on a separate page.
Do you have ghostscript?

At least, I hope that is the problem.

**Comment by hhrutter** [comment]
The size issue after split is actually already fixed!
